### PR TITLE
Remove test jars usage

### PR DESCRIPTION
**Pull request by geofjamg**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Technical debt

**What is the current behavior?** *(You can also link to an open issue here)*
We use test jars in many places to share classes between different artifact unit tests.

**What is the new behavior (if this is a feature change)?**
We don't use test jars anymore but normal jar except for TCK. This will allow easier migration to Java module system.
TCK: to be done later

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [x] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: psse/psse-converter/src/test/resources/ExampleVersion32.raw [hunk 1]
```diff
@@ -0,0 +1,58 @@
+ 0,   100.00,  32, 0, 0, 60.00     / PSS(R)E 32 RAW JUL 22 2014  15:40
```
[the remark this answers is not in the document]
Comment (geofjamg) [reply psse/psse-converter/src/test/resources/ExampleVersion32.raw]: I moved common test files to psse-converter-model

File: time-series/time-series-dsl/src/test/java/com/powsybl/timeseries/dsl/TestTimeSeriesIndex.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import com.powsybl.timeseries.AbstractTimeSeriesIndex;
+
+import java.time.Instant;
+import java.util.Iterator;
+import java.util.stream.Stream;
+
+/**
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+public class TestTimeSeriesIndex extends AbstractTimeSeriesIndex {
```
[the remark this answers is not in the document]
Comment (geofjamg) [reply time-series/time-series-dsl/src/test/java/com/powsybl/timeseries/dsl/TestTimeSeriesIndex.java]: I removed it and use `RegularTimeSeriesIndex` instead.

**Review by miovd — COMMENTED** [review]
Some propositions of renaming:
- `AbstractTestGridModel` -> `AbstractGridModel`
- `FakeCgmesModel` -> `InMemoryCgmesModel`
- `FakeTripleStore` -> `EmptyTripleStore`
- `TestGridModel` -> `GridModel`
- `TestGridModelResources` -> `ResourceGridModel`

**Comment by geofjamg** [comment]
> Some propositions of renaming:
> 
> * `AbstractTestGridModel` -> `AbstractGridModel`
> * `FakeCgmesModel` -> `InMemoryCgmesModel`
> * `FakeTripleStore` -> `EmptyTripleStore`
> * `TestGridModel` -> `GridModel`
> * `TestGridModelResources` -> `ResourceGridModel`

Why not GridModelReference (and AbstractGridModelReference, ResourceGridModelReference) as it embed a reference model that will be compared with another one, in case of TestGridModelResources to a one store in resources?

EmptyTripleStore: Ok
InMemoryCgmesModel: Ok

**Comment by geofjamg** [comment]
> > Some propositions of renaming:
> > 
> > * `AbstractTestGridModel` -> `AbstractGridModel`
> > * `FakeCgmesModel` -> `InMemoryCgmesModel`
> > * `FakeTripleStore` -> `EmptyTripleStore`
> > * `TestGridModel` -> `GridModel`
> > * `TestGridModelResources` -> `ResourceGridModel`
> 
> Why not GridModelReference (and AbstractGridModelReference, ResourceGridModelReference) as it embed a reference model that will be compared with another one, in case of TestGridModelResources to a one store in resources?
> 
> EmptyTripleStore: Ok InMemoryCgmesModel: Ok

Or maybe GridModelTestCase as it is only used in catalogs ?

**Comment by miovd** [comment]
> > Some propositions of renaming:
> > 
> > * `AbstractTestGridModel` -> `AbstractGridModel`
> > * `FakeCgmesModel` -> `InMemoryCgmesModel`
> > * `FakeTripleStore` -> `EmptyTripleStore`
> > * `TestGridModel` -> `GridModel`
> > * `TestGridModelResources` -> `ResourceGridModel`
> 
> Why not GridModelReference (and AbstractGridModelReference, ResourceGridModelReference) as it embed a reference model that will be compared with another one, in case of TestGridModelResources to a one store in resources?
> 
> EmptyTripleStore: Ok InMemoryCgmesModel: Ok

Yes, your propositions are better indeed

**Comment by miovd** [comment]
> > > Some propositions of renaming:
> > > 
> > > * `AbstractTestGridModel` -> `AbstractGridModel`
> > > * `FakeCgmesModel` -> `InMemoryCgmesModel`
> > > * `FakeTripleStore` -> `EmptyTripleStore`
> > > * `TestGridModel` -> `GridModel`
> > > * `TestGridModelResources` -> `ResourceGridModel`
> > 
> > 
> > Why not GridModelReference (and AbstractGridModelReference, ResourceGridModelReference) as it embed a reference model that will be compared with another one, in case of TestGridModelResources to a one store in resources?
> > EmptyTripleStore: Ok InMemoryCgmesModel: Ok
> 
> Or maybe GridModelTestCase as it is only used in catalogs ?

I prefer `GridModelReference`

**Comment by geofjamg** [comment]
> * `GridModel`

Done

**Review by miovd — COMMENTED** [review]
You should change `FakeCgmesModel` and `FakeTriplestore` too!

**Comment by geofjamg** [comment]
> You should change `FakeCgmesModel` and `FakeTriplestore` too!

Oups sorry I forgot to push

**Review by miovd — APPROVED** [review]

**Pull request merged**